src/python/numpyArrayView.hpp

```diff
@@ -0,0 +1,46 @@
+/*
+ *~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * Copyright (c) 2019, Lawrence Livermore National Security, LLC.
+ *
+ * Produced at the Lawrence Livermore National Laboratory
+ *
+ * LLNL-CODE-746361
+ *
+ * All rights reserved. See COPYRIGHT for details.
+ *
+ * This file is part of the GEOSX Simulation Framework.
+ *
+ * GEOSX is a free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License (as published by the
+ * Free Software Foundation) version 2.1 dated February 1999.
+ *~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ */
+
+/**
+ * @file numpyArrayView.hpp
+ */
+
+#pragma once
+#include "numpyConversion.hpp"
+#include "../ArrayView.hpp"
+
+namespace LvArray
+{
+
+namespace python
+{
+
+/**
+ * Return a Numpy view of an Array
+ * @param arr the ArrayView to convert to numpy.
```

Comment:
Document the template parameters with `@tparam` and add a `@brief` describing what you can and can't do with the returned array. For example you can't resize it and if `T` is `const` you can't assign to it either.

Reply:
What would happen if the caller resizes the array? It seems like the original `Array` would be invalid. Can we add a validity check for the variables that were provided to python upon calling the c-code again?

Reply:
I'm not sure what you mean.

NumPy prevents you from resizing externally allocated arrays so we're safe if someone in python tries to resize a NumPy array they got from geosx.

And we agreed that it is invalid to keep references to objects obtained from geosx around in-between time steps. There may be a way to keep track of all the objects that geosx exports into python and invalidate them each time it does a time step but it wouldn't be pretty.

Reply:
Well is you cannot resize an externally allocated array in numpy, then it is a pointless request. I read your comment incorrectly as a "shouldn't resize" instead of "can't resize". 

So what can you and can't you do on the python side? 
You CANNOT reallocate. 
Can you modify the data even if the source is `ArrayView< T const >`?

Reply:
You cannot reallocate and if `T` is const you can't modify the values.

Reply:
ok. I read the intent of your entire request incorrectly.

Reply:
You get reasonably nice errors:

```
62: Traceback (most recent call last):
62:   File "/g/g12/corbett8/LvArray/unitTests/python/testPythonSortedArrayDriver.py", line 47, in test_resizing
62:     array_from_c.resize((100,))
62: ValueError: cannot resize this array: it does not own its data
```

```
62: Traceback (most recent call last):
62:   File "/g/g12/corbett8/LvArray/unitTests/python/testPythonSortedArrayDriver.py", line 38, in test_modification
62:     array_from_c *= 2
62: ValueError: output array is read-only
```